xarray/tests/test_dataset.py

```diff
@@ -1392,6 +1392,14 @@ def test_sel_dataarray_mindex(self):
                 )
             )
 
+    def test_sel_categorical(self):
+        ind = pd.Series(["foo", "bar"], dtype="category")
+        df = pd.DataFrame({"ind": ind, "values": [1, 2]})
+        ds = df.set_index("ind").to_xarray()
+        actual = ds.sel(ind="bar")
+        expected = ds.isel(ind=1)
+        assert_identical(expected, actual)
```

Comment:
can we add a test for the error raised when `tolerance` is passed

Reply:
Thanks. Done